### PR TITLE
chore: fix multiple documentation names and links

### PR DESCRIPTION
**Pull request by i4ki**

### CODE REVIEW

File: docs/functions/tm_flatten.md [hunk 1]
```diff
@@ -25,80 +25,6 @@ tm_flatten([[["a", "b"], []], ["c"]])
 
 Indirectly-nested lists, such as those in maps, are _not_ flattened.
 
-## Flattening nested structures for `for_each`
```
Comment (i4ki) [inline docs/functions/tm_flatten.md:28]: those examples uses the Terraform `for_each` and do not make sense for Terramate.

File: docs/functions/tm_list.md
Comment (i4ki) [inline docs/functions/tm_list.md:1]: no longer available function

File: docs/functions/tm_map.md
Comment (i4ki) [inline docs/functions/tm_map.md:1]: old function, no longer available.

File: docs/functions/tm_setproduct.md [hunk 2]
```diff
@@ -112,161 +112,6 @@ tm_setproduct(["staging", "production"], ["a", 2])
 ]
 ```
 
-## Finding combinations for `for_each`
```
Comment (i4ki) [inline docs/functions/tm_setproduct.md:115]: more cases of Terraform examples

File: docs/functions/tm_type.md
Comment (i4ki) [inline docs/functions/tm_type.md:1]: this is function is not available in Terramate

**Review by soerenmartius — APPROVED** [review]
Awesome job @i4ki! Thanks a lot!

**Pull request merged**